smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/protocols/ProtocolPriorityConfig.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package software.amazon.smithy.typescript.codegen.protocols;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import software.amazon.smithy.model.shapes.ShapeId;
+
+
+/**
+ * Allows customization of protocol selection for specific services or a global default ordering.
+ */
+public final class ProtocolPriorityConfig {
```

Comment:
Can we make `ProtocolPriorityConfig`  immutable? Other than in the test, are there any other scenarios in which we need to mutate this after reading?

Reply:
made the class immutable, but settings can still overwrite the entire object